torch/_inductor/kernel/mm_scaled_grouped.py

```diff
@@ -167,46 +123,127 @@ def early_config_prune(configs, named_args):
 
 # Copied from fbgemm grouped_gemm.py
 triton_scaled_grouped_mm_source = r"""
-{{def_kernel("a_ptr", "b_ptr", "a_scale_ptr", "b_scale_ptr", "m_sizes")}}
+{% if A_IS_2D or B_IS_2D %}
```

Comment:
I was kind of contemplating whether it would be better to have a separate kernel for each 2d/3d case, since nested control flow in jinja templates is kind of hard to read.  But it would come with the downsides of some amount of code duplication.  Just curious to hear your opinion.

Reply:
For me, it's just a matter of a trade-off...   It seems to me the differences between the variations are not that big to warrant separate kernels, it would be too much code duplication.  Also, non-scaled version could be easily added into the same code, the same way (which I'm going to do next).   Furthermore, the Jinja if-else statements make the generated code readable (IMO), basically this generated code is like a separated version for given 2d/3d case.   There are only two cases of nested if-else Jinja statements; I could use some kind of indentation to make it more clear.   Overall, I'd prefer to keep it as is throughout the development, as for me it's easier to work with it this way; when we're happy with it, we could arrange it differently.

Reply:
Would be cool if torchtitan (that  has eager versions of these kernels) also adopted a single kernel approach, cc @lessw2020